### PR TITLE
std.os: fix uname usage.

### LINKED ISSUE #15839: MacOS (arm64): root struct of file 'c' has no member named 'utsname'
### Zig Version

0.11.0-dev.3220+447a30299

## Reproduce
Run the following program:

```zig
const std = @import("std");

pub fn main() !void {
    const stdout_file = std.io.getStdOut().writer();
    var bw = std.io.bufferedWriter(stdout_file);
    const stdout = bw.writer();

    try stdout.print("{}\n", .{std.os.uname()});

    try bw.flush();
}
```

## Result

```
zig build run
zig build-exe hello-world Debug native: error: the following command failed with 1 compilation errors:
/Users/simon/zig/zig/build/stage3/bin/zig build-exe /Users/simon/zig/hello-world/src/main.zig --cache-dir /Users/simon/zig/hello-world/zig-cache --global-cache-dir /Users/simon/.cache/zig --name hello-world --listen=-
Build Summary: 0/5 steps succeeded; 1 failed (disable with -fno-summary)
run transitive failure
└─ run hello-world transitive failure
   ├─ zig build-exe hello-world Debug native 1 errors
   └─ install transitive failure
      └─ install hello-world transitive failure
         └─ zig build-exe hello-world Debug native (reused)
/Users/simon/zig/zig/build/stage3/lib/zig/std/os.zig:195:27: error: root struct of file 'c' has no member named 'utsname'
pub const utsname = system.utsname;
                    ~~~~~~^~~~~~~~
referenced by:
    uname: /Users/simon/zig/zig/build/stage3/lib/zig/std/os.zig:5745:16
    main: src/main.zig:19:51
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

Something like this:
```
utsname{ .sysname = Darwin, .nodename = xxx, .release = 5.8.12_1, .version = xxxx .machine = x86_64, .domainname = xxxx }
```

### PR DESCRIPTION
**Pull request by devnexen**
close #15839.

### CODE REVIEW

File: lib/std/c/darwin.zig [hunk 1]
```diff
@@ -4155,3 +4155,15 @@ pub const vm_extmod_statistics_t = *vm_extmod_statistics;
 pub const vm_extmod_statistics_data_t = vm_extmod_statistics;
 
 pub extern "c" fn vm_stats(info: ?*anyopaque, count: *c_uint) kern_return_t;
+
+pub const _SYS_NAMELEN = 256;
+
+pub const utsname = extern struct {
+    sysname: [_SYS_NAMELEN:0]u8,
```
Comment (Vexu) [inline lib/std/c/darwin.zig:4162]: These arrays are 257 bytes long (_SYS_NAMELEN 256 + sentinel 1).

Comment (voltflake) [reply lib/std/c/darwin.zig]: See: https://www.man7.org/linux/man-pages/man2/uname.2.html
Third paragraph in notes section
> The length of the fields in the struct varies.  Some operating
> systems or libraries use a hardcoded 9 or 33 or 65 or 257.  Other
> systems use SYS_NMLN or _SYS_NMLN or UTSLEN or _UTSNAME_LENGTH.
> Clearly, it is a bad idea to use any of these constants; just use
> sizeof(...).  Often 257 is chosen in order to have room for an
> internet hostname.

Comment (Vexu) [reply lib/std/c/darwin.zig]: 256 is correct for macos based on https://github.com/ziglang/zig/blob/41502c6aa53a3da31b276c23c4db74db7d04796b/lib/libc/include/any-macos-any/sys/utsname.h#L72-L80

**Comment by devnexen** [comment]
cc @Vexu 🙂

**Pull request merged**